docs/examples/hello_tf_job_api.rst

```diff
@@ -0,0 +1,221 @@
+
+.. code-block:: shell
+
+  (nvflare-env) $ python3 -m pip install tensorflow
+
+With all the required dependencies installed, you are ready to run a Federated Learning system
+with two clients and one server. If you would like to go ahead and run the exercise now, you can run
+the ``fedavg_script_executor_hello-tf.py`` script which builds the job with the Job API and runs the
+job with the FLARE Simulator.
+
+NVIDIA FLARE Job API
+--------------------
+The ``fedavg_script_executor_hello-tf.py`` script for this hello-tf example is very similar to the ``fedavg_script_executor_hello-numpy.py`` script
+for the :doc:`Hello FedAvg with NumPy <hello_fedavg_w_numpy>` example and also the script for the :doc:`Hello PyTorch <hello_pt>`
+example. Other than changes to the names of the job and client script, the only difference is the line to define the initial global model
+for the server:
+
+.. code-block:: python
+
+   # Define the initial global model and send to server
+   job.to(TFNet(), "server")
+
+
+NVIDIA FLARE Client Training Script
+------------------------------------
+The training script for this example, ``hello-tf_fl.py``, is the main script that will be run on the clients. It contains the TensorFlow specific
+logic for training.
+
+Neural Network
+^^^^^^^^^^^^^^^
+Let's see what a simplified MNIST network looks like.
+
+.. literalinclude:: ../../examples/hello-world/hello-tf/src/tf_net.py
+   :language: python
+   :lines: 15-
+   :lineno-start: 15
+   :linenos:
+   :caption: tf_net.py
+
+This ``TFNet`` class is the convolutional neural network to train with MNIST dataset.
+This is not related to NVIDIA FLARE, and it is implemented in a file called ``tf_net.py``.
+
+Dataset & Setup
+^^^^^^^^^^^^^^^^
+Before starting training, you need to set up your dataset.
+In this exercise, it is downloaded from the Internet via ``tf.keras``'s datasets module
+and split in half to create a separate dataset for each client.
+Additionally, the optimizer and loss function need to be configured.
+
+All of this happens before the ``while flare.is_running():`` line in ``hello-tf_fl.py``.
+
+.. literalinclude:: ../../examples/hello-world/hello-tf/src/hello-tf_fl.py
+   :language: python
+   :lines: 29-57
+   :lineno-start: 29
+   :linenos:
+   :caption: hello-tf_fl.py
+
+Client Local Train
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The client code gets the weights from the input_model received from the server then performs a simple :code:`self.model.fit`
+so the client's model is trained with its own dataset:
+
+.. literalinclude:: ../../examples/hello-world/hello-tf/src/hello-tf_fl.py
+   :language: python
+   :lines: 58-91
+   :lineno-start: 58
+   :linenos:
+  
+After finishing the local train, the train method uses the newly-trained weights to build a new ``DXO`` to update the
+``Shareable`` with and then returns it back to the NVIDIA FLARE server.
```

Comment:
Do we want to mention DXO and Shareable here where are we showing them ?

Reply:
I copied this from the previous example without updating, I will update this